lib/daemon.js

```diff
@@ -36,7 +36,7 @@ class Daemon {
    * @returns {Object} Poller descriptor.
    */
   _createPollerDescriptor (externalSecret) {
-    const { uid, name, namespace } = externalSecret.metadata
+    const { uid, name, namespace, annotations = {} } = externalSecret.metadata
```

Comment:
In general, I worry about copying one object's annotations directly to another object.  A more "kubernetes-like" approach would be to have a `Secret` template (analogous to the `Pod` template for `Deployment`s) and set the annotation values there that you want the external secrets controller to apply to the corresponding `Secret`.

We've discussed this approach before, but had some concerns with reconciling in with other "frontends" that don't use `Secret` objects (see https://github.com/godaddy/kubernetes-external-secrets/issues/46).

What's the use case for this?  Maybe there's another way to support it?